### ISSUE TITLE
Assign missing codes to PyFlakes messages - [merged]

### ISSUE THREAD

**Issue by asottile** [PyCQA/flake8#832]
In GitLab by @ambv on Nov 27, 2016, 17:44

_Merges pyflakes_missing_codes -> master_

Some PyFlakes messages weren't covered by unique messages, making them
impossible to select/ignore. This is now fixed.

To ensure we don't regress in the future, a test has been added that fails if
there's any uncovered messages.

**Comment by asottile** [comment]
In GitLab by @pycqa-jenkins on Nov 27, 2016, 17:46

:-1: Jenkins Build Failed

Results available at: [Jenkins [Flake8 Build Pipeline #217]](https://ci.pycqa.org/job/flake8-trigger/217/)

**Comment by asottile** [comment]
In GitLab by @pycqa-jenkins on Nov 27, 2016, 17:47

:-1: Jenkins Build Failed

Results available at: [Jenkins [Flake8 Build Pipeline #218]](https://ci.pycqa.org/job/flake8-trigger/218/)

**Comment by asottile** [comment]
In GitLab by @ambv on Nov 27, 2016, 17:51

'linters' and 'dogfood' are failing with a missing code coverage report: https://gitlab.com/ambv/flake8/builds/6715620

I don't understand this.

**Comment by asottile** [comment]
In GitLab by @ambv on Nov 27, 2016, 18:16

Added 1 commit:

* 6cb01378 - Assign missing codes to PyFlakes messages

[Compare with previous version](https://gitlab.com/pycqa/flake8/merge_requests/159/diffs?diff_id=1606869&start_sha=e085970579b69e8c07b46b7d6c669814d81f9a54)

**Comment by asottile** [comment]
In GitLab by @ambv on Nov 27, 2016, 18:17

Alright, the actual problem were import lint warnings. Fixed now, passed tox on 2.7.11, 3.3.6, 3.4.5, 3.5.2.

**Comment by asottile** [comment]
In GitLab by @pycqa-jenkins on Nov 27, 2016, 18:18

:+1: Jenkins Build Success

Results available at: [Jenkins [Flake8 Build Pipeline #219]](https://ci.pycqa.org/job/flake8-trigger/219/)

**Comment by asottile** [comment]
In GitLab by @sigmavirus24 on Nov 28, 2016, 06:07

Commented on [tests/unit/test_pyflakes_codes.py line 1](https://github.com/PyCQA/flake8/compare/b8ce1334d099aee9c7d2ea72f75bf01ccbe2b040..6cb013788f5d6e7f9a49d3ff569678f9f7f56837#diff-1152aba171f0ceb143e27461625bf0a4R1)



This docstring doesn't seem to match.

**Comment by asottile** [comment]
In GitLab by @sigmavirus24 on Nov 28, 2016, 06:08

Commented on [tests/unit/test_pyflakes_codes.py line 5](https://github.com/PyCQA/flake8/compare/b8ce1334d099aee9c7d2ea72f75bf01ccbe2b040..6cb013788f5d6e7f9a49d3ff569678f9f7f56837#diff-1152aba171f0ceb143e27461625bf0a4R5)



The project prefers importing modules instead of objects. For example,

```py
from flake8.plugins import pyflakes as pyflakes_shim
```

**Comment by asottile** [comment]
In GitLab by @sigmavirus24 on Nov 28, 2016, 06:08

Commented on [tests/unit/test_pyflakes_codes.py line 16](https://github.com/PyCQA/flake8/compare/b8ce1334d099aee9c7d2ea72f75bf01ccbe2b040..6cb013788f5d6e7f9a49d3ff569678f9f7f56837#diff-1152aba171f0ceb143e27461625bf0a4R16)



Since we use PyTest, you can more simply do:

```
assert messages == set(pyflakes_shim.FLAKE8_PYFLAKES_CODES)
```

And it will automagically provide the diff.

**Comment by asottile** [comment]
In GitLab by @sigmavirus24 on Nov 28, 2016, 06:09

Could you also add a note to the 3.3.0 release notes mentioning these new error codes?

**Comment by asottile** [comment]
In GitLab by @ambv on Nov 28, 2016, 09:35

Commented on [tests/unit/test_pyflakes_codes.py line 5](https://github.com/PyCQA/flake8/compare/b8ce1334d099aee9c7d2ea72f75bf01ccbe2b040..6cb013788f5d6e7f9a49d3ff569678f9f7f56837#diff-1152aba171f0ceb143e27461625bf0a4R5)



Alright. Will fix. What's the rationale and where is this documented?

**Comment by asottile** [comment]
In GitLab by @ambv on Nov 28, 2016, 09:36

Commented on [tests/unit/test_pyflakes_codes.py line 16](https://github.com/PyCQA/flake8/compare/b8ce1334d099aee9c7d2ea72f75bf01ccbe2b040..6cb013788f5d6e7f9a49d3ff569678f9f7f56837#diff-1152aba171f0ceb143e27461625bf0a4R16)



Will fix.

**Comment by asottile** [comment]
In GitLab by @ambv on Nov 28, 2016, 09:36

Sure.

**Comment by asottile** [comment]
In GitLab by @ambv on Nov 28, 2016, 10:52

Added 1 commit:

* 2f1338c3 - Assign missing codes to PyFlakes messages

[Compare with previous version](https://gitlab.com/pycqa/flake8/merge_requests/159/diffs?diff_id=1617341&start_sha=6cb013788f5d6e7f9a49d3ff569678f9f7f56837)

**Comment by asottile** [comment]
In GitLab by @ambv on Nov 28, 2016, 10:54

Alright, all fixed and Tox tested.

**Comment by asottile** [comment]
In GitLab by @pycqa-jenkins on Nov 28, 2016, 10:54

:+1: Jenkins Build Success

Results available at: [Jenkins [Flake8 Build Pipeline #220]](https://ci.pycqa.org/job/flake8-trigger/220/)

**Comment by asottile** [comment]
In GitLab by @sigmavirus24 on Nov 29, 2016, 04:56

Commented on [tests/unit/test_pyflakes_codes.py line 5](https://github.com/PyCQA/flake8/compare/b8ce1334d099aee9c7d2ea72f75bf01ccbe2b040..6cb013788f5d6e7f9a49d3ff569678f9f7f56837#diff-1152aba171f0ceb143e27461625bf0a4R5)



Good point. This *needs* documenting. This is mostly my personal preference but I find the imports are usually:

- Simpler and less complex when done once per-line and by only importing modules
- It's easier to find where a function/constant/class came from when reading the code where it's used if the module is prefixing it (rather than scanning to the top of the file or using something like ctags integrated with my editor)

tl;dr This is a part of the Google Style Guide that I appreciate having worked on OpenStack and seen the above benefits manifested there.

**Comment by asottile** [comment]
In GitLab by @sigmavirus24 on Nov 29, 2016, 05:01

Commented on [tests/unit/test_pyflakes_codes.py line 5](https://github.com/PyCQA/flake8/compare/b8ce1334d099aee9c7d2ea72f75bf01ccbe2b040..6cb013788f5d6e7f9a49d3ff569678f9f7f56837#diff-1152aba171f0ceb143e27461625bf0a4R5)



Created an issue for this discussion: https://gitlab.com/pycqa/flake8/issues/276

**Comment by asottile** [comment]
In GitLab by @sigmavirus24 on Nov 29, 2016, 05:01

Resolved all discussions

**Comment by asottile** [comment]
In GitLab by @sigmavirus24 on Nov 29, 2016, 05:02

Mentioned in commit 348722d77b32ad2ecc064559bb8b4cae08e85549
